egui_demo_lib/src/apps/demo/paint_bezier.rs

```diff
@@ -0,0 +1,182 @@
+use egui::*;
+use egui::emath::RectTransform;
+use egui::{epaint::{QuadraticBezierShape, CubicBezierShape,CircleShape}};
+
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(default))]
+pub struct PaintBezier {
+    bezier: usize, // current bezier curve degree, it can be 3,4,
+    bezier_backup: usize, //track the bezier degree before change in order to clean the remaining points.
+    points: Vec<Pos2>, //points already clicked. once it reaches the 'bezier' degree, it will be pushed into the 'shapes'
+    backup_points:Vec<Pos2>, //track last points set in order to draw auxiliary lines.
+    quadratic_shapes: Vec<QuadraticBezierShape>, //shapes already drawn. once it reaches the 'bezier' degree, it will be pushed into the 'shapes'
+    cubic_shapes: Vec<CubicBezierShape>,
```

Comment:
These can be combined into one `Vec<Shape>`

Reply:
Tried but gave up because `Shape` didn't implement `Serialization` and `Deserialization` from `Serde`. I don't want to add `#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]` on top of `Shape` since I don't know what's the impact.

Reply:
Oh, `Shape` should have `#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]` - I just forgot about it, likely.